src/authStrategies/RemoteAuth.js

```diff
@@ -0,0 +1,254 @@
+
+            /* Extracting Session From DB to Local Directory */
+            if (sessionExists) {
+                console.log('> Extracting Session...');
+                /* 2. await DB extraction to desired path */
+
+                /* Delete Previous local session folder */
+                if (fs.existsSync(this.userDataDir)) {
+                    await fsPromise.rm(this.userDataDir, {
+                        recursive: true,
+                        force: true
+                    });
+                }
+
+                await this.unzipSession('ExtractedSession.zip');
+                await fsPromise.unlink('ExtractedSession.zip');
+
+                console.log('> Extraction Completed');
+            }
+        } catch (error) {
+            console.log('RemoteAuth: ExtractSession =>', error);
+        }
+    }
+
+    static async deleteSession() {
+        try {
+            /**
+             * TODO:
+             * 1. @const {boolean} sessionExists => Call abstract DB interface to get session instance if exists
+             * 2. Call abstract DB interface to delete session
+             */
+             const sessionExists = false; /* 1. Default to false on initial draft */
+    
+            /* Deleting Session */
+            if (sessionExists) {
+                console.log('> Deleting Session...')
+                /* 2. await delete session from DB */
+    
+                console.log('Deleting Complete');
+            }
+        } catch (error) {
+            console.log('Error MD => DeleteSession', error);
+        }
+    }
+
+    /**
+     * @param {String} sourceDir
+     * @param {String} dirPath 
+     * @returns {Promise}
+     */
+    static async zipSession(sourceDir, dirPath) {
+        try {
+            const archive = archiver('zip');
+            const stream = fs.createWriteStream(dirPath);
+
+            await this.deleteMetadata();
+            return new Promise((resolve, reject) => {
+                archive
+                    .directory(sourceDir, false)
+                    .on('error', err => reject(err))
+                    .pipe(stream);
+
+                stream.on('close', () => resolve());
+                archive.finalize();
+            });
+        } catch (error) {
+            console.log('RemoteAuth: zipSession => ', error);
+        }
+    }
+
+    /**
+     * @param {String} sourceDir 
+     * @returns {Promise}
+     */
+    static async unzipSession(sourceDir) {
+        try {
+            var stream = fs.createReadStream(sourceDir);
+            return new Promise((resolve, reject) => {
+                stream.pipe(unzipper.Extract({
+                        path: this.userDataDir
+                    }))
+                    .on('error', err => reject(err))
+                    .on('finish', () => resolve());
+            });
+        } catch (error) {
+            console.log('RemoteAuth: unzipSession => ', error);
+        }
+    }
+
+    static async deleteMetadata() {
```

Comment:
I'm a little concerned that deleting the other files on a running instance might conflict with other things. Maybe we could use the options available in `tar` to filter and only include the files we need in the archive, or copy them to a temp directory.